scripts/linux/smokeTestHelper.sh

```diff
@@ -0,0 +1,220 @@
+  #    Note: To download a specific version, the ($1) can be specified as "aziot-edge=1.2.10-1"
+  # $2 - target output directory
+
+  pkgName=${1%=*}
+  uri=$(apt-get install --reinstall --print-uris -qq $1 | cut -d"'" -f2 | grep "/$pkgName/")
+  [[ -z "$uri" ]] && { echo "[FAIL] Package ($1) cannot be found in a known linux repository"; exit 1;}
+  artifactName=${uri##*/}
+  targetArtifactPath="$(realpath $2)/${artifactName%*.deb}_pmc.deb"
+  
+  cmd="wget -q $uri -O $targetArtifactPath"
+  echo "Running: $cmd"
+  $cmd
+}
+
+wait-for-dpkg-lock()
+{
+  # Wait for task 'apt.systemd.daily' to be completed. This is helpful if you later
+  # need to use `apt`. 
+  #
+  # $1 - wait time in seconds
+
+  waitAttempts=$1
+  sleepDuration=10
+  for (( i=0; $i<$waitAttempts; i++ ))
+  do
+    if [[ $(ps aux | grep -c apt.systemd.daily) == 1 ]]; then
+      break
+    fi
+    sleep $sleepDuration
+  done
+}
+
+setup-focal-source-apt()
+{
+  # 1ES image for Ubuntu20.04 (Focal) actually has a misconfigured apt repo source
+  # This function correct the repo source by directly configure the apt to point to 
+  # the Microsoft Linux Repository for Focal.
+
+  for file in $(ls /etc/apt/sources.list.d/*.list)
+  do
+    sudo sed -i 's/bionic/focal/g' $file
+  done
+
+  sudo sed -i 's/bionic/focal/g' /etc/apt/sources.list && \
+  sudo rm -f /etc/apt/sources.list.d/microsoft-prod.list && \
+  wget https://packages.microsoft.com/config/ubuntu/20.04/prod.list  -O /etc/apt/sources.list.d/microsoft-prod.list && \
+  sudo apt update -y && \
+  sudo DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" dist-upgrade && \
+  sudo apt update -y && \
+  sudo apt upgrade -y
+}
+
+setup-config-apt()
+{
+  # Download the `apt` config file from the provided uri ($1), install it, and update the apt
+  #
+  # $1 - uri for the config deb package
+  # Ref: https://docs.microsoft.com/en-us/azure/iot-edge/how-to-provision-single-device-linux-symmetric?view=iotedge-2020-11&tabs=azure-portal%2Cubuntu
+
+  # For ARM machine, we need to wait for the apt-get daily update to be done first before we can proceed.
+  wait-for-dpkg-lock 120
+
+  echo "Setup source artifact repository:"
+  echo "$1"
+  wget "$1" -O packages-microsoft-prod.deb &&
+  sudo dpkg --force-confnew -i packages-microsoft-prod.deb &&
+  rm packages-microsoft-prod.deb &&
+  sudo apt-get update -y
+}
+
+test-released-metadata()
+{
+  # Amalgam of test cases to test PMC or GitHub release page.
+  #
+  # $1 - package name (string)
+  # $2 - version string (string)
+
+  pkgName=$1
+  version=$2
+
+  echo "Get the latest version string from linux repository"
+  # Note: $latestVersionApt includes the trailing revision: "1.2.10-1"
+  latestVersionApt=$(get-latest-version-apt "$pkgName" 1)
+  check-matching-version "$latestVersionApt" "$version" "$pkgName"
+}
+
+test-released-artifact()
+{
+  # Amalgam of test cases to test PMC or GitHub release page.
```

Comment:
nit: would help readability to list test cases

Reply:
Sure thing. I'll address it in the next PR for the release package.